lib/broccoli/ember-app.js

```diff
@@ -341,7 +341,15 @@ EmberApp.prototype._notifyAddonIncluded = function() {
   this.project.addons = this.project.addons.filter(function(addon) {
     addon.app = this;
 
-    if (!addon.isEnabled || addon.isEnabled()) {
+    var addonEnabled = !addon.isEnabled || addon.isEnabled();
+
+    if (addonEnabled && this.options.addonBlacklist && this.options.addonBlacklist.indexOf(addon.name) !== -1) {
```

Comment:
lots of complexity in these lines, let me recommend this be extract to its own function, which then itself can be nicely unit tested.